### PR TITLE
Added support for filtering exported hosts

### PR DESCRIPTION
**Pull request by rmsds**
Allows for attaching arbitrary tags to exported resources, and collecting only exported resources matching a specific tag

Signed-off-by: Ricardo Silva ricardo.silva@epfl.ch

### CODE REVIEW

**Comment by rmsds** [comment]
Had the requirement for this feature independently, but this should address #38 .

File: manifests/init.pp [hunk 1]
```diff
@@ -121,16 +133,29 @@
   }
 
   if $use_fqdn_real == true {
-    @@host { $::fqdn:
-      ensure       => $fqdn_ensure,
-      host_aliases => $my_fqdn_host_aliases,
-      ip           => $fqdn_ip,
+    if $export_tag == undef {
+      @@host { $::fqdn:
```
Comment (ghoneycutt) [inline manifests/init.pp]: We should not duplicate the resource block, instead try this.

```puppet
@@host { $::fqdn:
  ensure       => $fqdn_ensure,
  host_aliases => $my_fqdn_host_aliases,
  ip           => $fqdn_ip,
  tag          => $export_tag,
}
```

File: manifests/init.pp [hunk 2]
```diff
@@ -4,6 +4,8 @@
 #
 class hosts (
   $collect_all           = false,
+  $collect_tag           = undef,
```
Comment (ghoneycutt) [inline manifests/init.pp]: spec tests are needed for the new params. They will need to show that the wrong types will fail.

File: manifests/init.pp (hunk 1 above)
Comment (rmsds) [inline manifests/init.pp]: I did try that at first, but it failed one test: Ruby 1.8.7 + Puppet Gem 3 w/ Future Parser (https://travis-ci.org/ghoneycutt/puppet-module-hosts/jobs/164903651)
Is there a better approach that will still pass that test?

Comment (ghoneycutt) [inline manifests/init.pp]: Aha, so you cannot have a tag on a resource that is undef or empty.

Could you please make a comment above that piece of code that explains this and why the code is duplicated.

**Comment by tuxmea** [comment]
Is there a reason why this PR is not yet merged? Is something missing?

**Comment by rmsds** [comment]
@tuxmea Not that I'm aware of. I've been using a version with these changes included for some time.

**Comment by ghoneycutt** [comment]
We can test for exported resources, which I'm doing in PR #61 though we cannot test for collected resources. I need to get Vagrant setup to ensure this works as expected.

**Comment by ghoneycutt** [comment]
Thank you everyone for your contributions and input. I've rebased this so the tags can be tested in PR #62

**Pull request closed without merging**